malariagen_data/util.py

```diff
@@ -1626,7 +1649,9 @@ def prep_samples_for_cohort_grouping(*, df_samples, area_by, period_by):
     return df_samples
 
 
-def build_cohorts_from_sample_grouping(*, group_samples_by_cohort, min_cohort_size):
+def build_cohorts_from_sample_grouping(
+    *, group_samples_by_cohort, min_cohort_size, taxon_by="taxon"
```

Comment:
```suggestion
    *, group_samples_by_cohort, min_cohort_size, taxon_by
```

Shouldn't need a default value here.

Reply:
This was needed while the other `..._frequencies_advanced()` functions didn't support the parameter, e.g. calls such as this, in `haplotypes_frequencies_advanced()` and `_gene_cnv_frequencies_advanced()`, where `taxon_by` is not otherwise specified:
```python
        df_cohorts = build_cohorts_from_sample_grouping(
            group_samples_by_cohort=group_samples_by_cohort,
            min_cohort_size=min_cohort_size,
        )
```
...which would result in a problem around this expectation of a `taxon_by ` value in `build_cohorts_from_sample_grouping()`:
```python
    df_cohorts["label"] = df_cohorts.apply(
        lambda v: f"{v.area}_{v[taxon_by][:4]}_{v.period}", axis="columns"
    )
```
...but it looks like this becomes  a non-issue when the other functions gain support, e.g.
```python
        df_cohorts = build_cohorts_from_sample_grouping(
            group_samples_by_cohort=group_samples_by_cohort,
            min_cohort_size=min_cohort_size,
            taxon_by=taxon_by,
        )
```